pkg/util/util.go

```diff
@@ -275,3 +276,11 @@ func GenerateProxySecret() (string, error) {
 	return base64Secret, nil
 
 }
+
+// TrimNonAlphaNumeric remove all non-alphanumeric values from start and end of the string
+func TrimNonAlphaNumeric(text string) string {
```

Comment:
I think this functionality needs to be included in the `Truncate` method, as will probably apply to all names that are truncated?

Also the `regexp.Compile` patterns should be statics, so only generated one.

Reply:
+1